src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package mate.service;
+
+import mate.dao.DriverDao;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverDao.getByLogin(login).orElseThrow(()
+                -> new AuthenticationException("Login or password is incorrect"));
+        if (driver.getPassword().equals(password)) {
+            return driver;
+        }
+        throw new AuthenticationException("Login or password is incorrect");
```

Comment:
throw exception only one time 
```suggestion
        Optional<Driver> driver = driverService.findByLogin(login);
        if (driver.isEmpty() || !driver.get().getPassword().equals(password)) {
            throw new AuthenticationException("Login or password was incorrect fo user: "+ login);
        }
        return driver.get();
```